### PR TITLE
Delete_User/delete_all_the_user_events

### PR DESCRIPTION
**Pull request by Lisafiluz**

### CODE REVIEW

**Comment by Lisafiluz** [comment]
include user deletion in UI + BE

File: app/routers/profile.py [hunk 1]
```diff
@@ -46,6 +60,12 @@ async def profile(
 
     session.close()
```
Comment (ivarshav) [inline app/routers/profile.py:61]: Are you sure you need to close the session?

File: app/routers/profile.py [hunk 2]
```diff
@@ -23,18 +24,31 @@
...
 @router.get("/")
 async def profile(
         request: Request,
         session=Depends(get_db),
-        new_user=Depends(get_placeholder_user)):
+        new_user=Depends(get_placeholder_user),
+        new_event=Depends(get_placeholder_event)):
 
     # Get relevant data from database
     upcouming_events = range(5)
```
Comment (ivarshav) [inline app/routers/profile.py:54]: Maybe instaed of hard-coded "5", make it configurable?

File: app/routers/profile.py [hunk 3]
```diff
@@ -108,6 +128,28 @@ async def update_profile(
     return RedirectResponse(url=url, status_code=HTTP_302_FOUND)
 
 
+async def get_current_user():
+    user = get_placeholder_user()
```
Comment (ivarshav) [inline app/routers/profile.py:132]: Consider `return get_placeholder_user()`.
I think that a documention and typing could help here.

File: app/routers/profile.py [hunk 4]
```diff
@@ -46,6 +60,12 @@ async def profile(
 
     session.close()
 
+    user_events = session.query(Event).filter_by(owner_id=user.id).all()
+    if not user_events:
+        session.add(new_event)
+        session.commit()
+        user_events = session.query(Event).filter_by(owner_id=user.id).first()
+    print(user_events)
```
Comment (ivarshav) [inline app/routers/profile.py:68]: remove print :)

File: app/routers/profile.py [hunk 5]
```diff
@@ -108,6 +128,28 @@ async def update_profile(
...
+async def delete_user(
+        request: Request, session=Depends(get_db), current_user: User = Depends(get_current_user)):
+    
+    # Delete all the events of the user
+    session.query(Event).filter_by(owner_id=current_user.id).delete()
+    session.commit()
+    session.close()
+    
+    # Delete the user
+    session.query(User).filter_by(username=current_user.username).delete()
```
Comment (ivarshav) [inline app/routers/profile.py:145]: You can't delete the user with a closed session

File: app/routers/profile.py [hunk 6]
```diff
@@ -108,6 +128,28 @@ async def update_profile(
...
+    session.commit()
+    session.close()
+
+    url = '/'
```
Comment (ivarshav) [inline app/routers/profile.py:149]: Maybe const with a better name to explain this scenario?

File: app/static/style.css [hunk 7]
```diff
@@ -60,4 +60,7 @@ body {
     border: none;
     background-color:whitesmoke;
 }
-  
\ No newline at end of file
+
```
Comment (ivarshav) [inline app/static/style.css:63]: Keep one empty line

File: tests/test_profile.py [hunk 8]
```diff
@@ -107,3 +107,13 @@ def test_upload_user_photo(profile_test_client):
...
+
+
+def test_delete_user_(profile_test_client):
+    # Get profile page and initialize database
+    profile = profile_test_client.get('/profile')
+
+    # Post delete user
+    response = profile_test_client.post(
+        '/profile/delete_user')
+    assert response.status_code == 302
```
Comment (ivarshav) [inline tests/test_profile.py:119]: use HTTP_302_FOUND

File: tests/test_profile.py [hunk 9]
```diff
@@ -107,3 +107,13 @@ def test_upload_user_photo(profile_test_client):
...
     new_avatar_path = os.path.join(MEDIA_PATH, 'fake_user.png')
     assert Image.open(new_avatar_path).size == config.AVATAR_SIZE
+
+
+def test_delete_user_(profile_test_client):
+    # Get profile page and initialize database
+    profile = profile_test_client.get('/profile')
+
+    # Post delete user
+    response = profile_test_client.post(
```
Comment (ivarshav) [inline tests/test_profile.py:117]: I  think it can be one line

File: app/templates/profile.html [hunk 10]
```diff
@@ -131,6 +138,25 @@ <h5 class="modal-title">Upload photo</h5>
...
+                                        <button type="button" class="btn-close" data-bs-dismiss="modal" aria-label="Close"></button>
+                                    </div>
+                                    <div class="modal-body">
+                                        <form action="delete_user" method="post">
+                                            <div class="d-flex flex-row-reverse mt-4">
+                                                <button type="submit" class="btn btn-lg btn-danger">Delete</button>
+                                            </div>
+                                        </form>
+                                    </div>
+
```
Comment (ivarshav) [inline app/templates/profile.html:156]: Why do you hace an empty line here?

File: app/routers/profile.py (hunk 2 above)
Comment (Lisafiluz) [reply app/routers/profile.py:54]: it is not my line, once the user will able to create events, probably this line will be deleted

File: app/routers/profile.py (hunk 1 above)
Comment (Lisafiluz) [reply app/routers/profile.py:61]: fixed

File: app/routers/profile.py [hunk 11]
```diff
@@ -23,18 +24,31 @@
 
 def get_placeholder_user():
     return User(
+        id=1,
```
Comment (efratush) [inline app/routers/profile.py:27]: If id = 1 is busy, the program will crash. 
Recommends a variable boot at first startup and then retrieving the information from there.

File: app/routers/profile.py [hunk 12]
```diff
@@ -108,6 +127,35 @@ async def update_profile(
...
+        session.query(User).filter_by(username=current_user.username).delete()
+        session.commit()
+        session.close()
+
+        # Redirect to the home page after the deletion
+        url = '/'
+        return RedirectResponse(url=url, status_code=HTTP_302_FOUND)
+    else:
+        url = router.url_path_for("profile")
+        return RedirectResponse(url=url, status_code=HTTP_302_FOUND)
```
Comment (efratush) [inline app/routers/profile.py:156]: How do you inform him that the deletion failed?

File: app/routers/profile.py [hunk 13]
```diff
@@ -23,18 +24,31 @@
...
+        title="new_event",
+        content="content_of_the_event",
+        start=datetime.date(2007, 5, 26),
+        end=datetime.date(2007, 5, 27),
+        owner_id=1
+    )
+
+
 @router.get("/")
 async def profile(
```
Comment (yammesicka) [inline app/routers/profile.py:47]: Please add type annotations in your code :)

File: app/routers/profile.py (hunk 1 above)
Comment (yammesicka) [reply app/routers/profile.py:61]: I still see it in the code. Forgot to push, maybe?

File: app/routers/profile.py [hunk 14]
```diff
@@ -108,6 +127,35 @@ async def update_profile(
     return RedirectResponse(url=url, status_code=HTTP_302_FOUND)
 
 
+async def get_current_user():
+    return get_placeholder_user()
+
+
+@router.post("/delete_user")
```
Comment (yammesicka) [inline app/routers/profile.py:134]: Maybe `@router.delete('/')`?

File: tests/test_profile.py [hunk 15]
```diff
@@ -107,3 +108,15 @@ def test_upload_user_photo(profile_test_client):
...
+
+def test_delete_user_(profile_test_client):
+    # Get profile page and initialize database
+    profile_test_client.get('/profile')
+    data = {
+        "username_confirmation": "fake_user"
+    }
+    # Post delete user
+    response = profile_test_client.post(
+        '/profile/delete_user', data=data)
```
Comment (yammesicka) [inline tests/test_profile.py:121]: Prefer `url_for`

File: tests/test_profile.py [hunk 16]
```diff
@@ -107,3 +108,15 @@ def test_upload_user_photo(profile_test_client):
...
+    assert response.status_code == HTTP_302_FOUND
```
Comment (yammesicka) [inline tests/test_profile.py:122]: Prefer `.ok`

**Review by yammesicka — CHANGES_REQUESTED** [review]
Great job! Added few of my insights.

**Comment by yammesicka** [comment]
Please do not upload the node_modules directory (add the [following lines](https://github.com/github/gitignore/blob/master/Node.gitignore) to the `.gitignore` file).

**Pull request closed without merging**